### ISSUE TITLE
Oxidized does not send username during SSH login

### ISSUE THREAD

**Issue by yostie** [ytti/oxidized#393]
I have about 78 nodes loaded and am having problems with 3 of them.  Two are Cisco 2960X's running regular IOS and the last is a 3850 running IOS-XE.  Oxidized will time out attempting to connect to them.  I have gone through many of the open/closed issues that sound similar to this but maybe not 100%.  The logs show:

```
D, [2016-04-12T21:15:46.479606 #14341] DEBUG -- : 10.10.10.10 raised Net::SSH::Disconnect with msg "connection closed by remote host"
W, [2016-04-12T21:15:47.379973 #14341]  WARN -- : device.example.com status no_connection, retry attempt 1
D, [2016-04-12T21:15:47.380214 #14341] DEBUG -- : Jobs 0, Want: 1
```

It does this for the timeout/retry values and then obviously moves on.  I can confirm that if I SSH from the device running Oxidized, it will connect and I can manually log in.  The one thing I did notice is if I login to the switch itself and do a "who" to see who is connecting, I see myself and I see an entry for a user with no username:

```
MY_HOSTNAME#who
    Line       User       Host(s)              Idle       Location
*   1 vty 0    dev_admin idle                 00:00:00 192.168.0.212
    2 vty 1              idle                 00:00:07 172.16.10.10

```

It's almost as if oxidized is not reading the username prompt correctly and is waiting to send the username.  If I SSH from SecureCRT, the prompt looks like:

```
Please enter login information for device.mydomain.com.
Username: 

```

Finally, if I SSH from the host running oxidized w/ verbose on:

```
[me@myhost oxidized]$ ssh -v dev_admin@device.mydomain.com
OpenSSH_6.6.1, OpenSSL 1.0.1e-fips 11 Feb 2013
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 56: Applying options for *
debug1: Connecting to device.mydomain.com [10.10.10.10] port 22.
debug1: Connection established.
debug1: identity file /home/me/.ssh/id_rsa type -1
debug1: identity file /home/me/.ssh/id_rsa-cert type -1
debug1: identity file /home/me/.ssh/id_dsa type -1
debug1: identity file /home/me/.ssh/id_dsa-cert type -1
debug1: identity file /home/me/.ssh/id_ecdsa type -1
debug1: identity file /home/me/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/me/.ssh/id_ed25519 type -1
debug1: identity file /home/me/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_6.6.1
debug1: Remote protocol version 2.0, remote software version Cisco-1.25
debug1: no match: Cisco-1.25
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: server->client aes128-ctr hmac-sha1 none
debug1: kex: client->server aes128-ctr hmac-sha1 none
debug1: kex: diffie-hellman-group-exchange-sha1 need=20 dh_need=20
debug1: kex: diffie-hellman-group-exchange-sha1 need=20 dh_need=20
debug1: SSH2_MSG_KEX_DH_GEX_REQUEST(1024<7680<8192) sent
debug1: expecting SSH2_MSG_KEX_DH_GEX_GROUP
debug1: SSH2_MSG_KEX_DH_GEX_INIT sent
debug1: expecting SSH2_MSG_KEX_DH_GEX_REPLY
debug1: Server host key: RSA 61:c0:ff:61:60:16:4f:a3:74:c6:68:72:8c:3c:9e:23
debug1: Host 'device.mydomain.com' is known and matches the RSA host key.
debug1: Found key in /home/me/.ssh/known_hosts:8
debug1: ssh_rsa_verify: signature correct
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: SSH2_MSG_SERVICE_REQUEST sent
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,keyboard-interactive,password
debug1: Next authentication method: publickey
debug1: Trying private key: /home/me/.ssh/id_rsa
debug1: Trying private key: /home/me/.ssh/id_dsa
debug1: Trying private key: /home/me/.ssh/id_ecdsa
debug1: Trying private key: /home/me/.ssh/id_ed25519
debug1: Next authentication method: keyboard-interactive
Password: 
debug1: Authentication succeeded (keyboard-interactive).
Authenticated to device.mydomain.com ([10.10.10.10]:22).
debug1: channel 0: new [client-session]
debug1: Entering interactive session.
debug1: Sending environment.
debug1: Sending env LANG = en_US.UTF-8

```

Not sure what the issue is.  I've tried using the oxs command and turning on -v and --terse but that doesn't give me anything useful.  I have also tried turning on input debugging in the config file, but no log file is actually created.

**Comment by ytti** [comment]
Which ssh version are you running. 'gem list | grep ssh'. There is another bug open about ssh issue with newer library against ciscos. I'm going to look into this hopefully very soon. It may be you're hitting the same.

**Comment by slinderud** [comment]
Hey,

I solved this issue by downgrading net-ssh to 3.0.1:
gem install net-ssh -v 3.0.1

Could you try this and report back? Might have to uninstall net-ssh 3.1.1 first

**Comment by yostie** [comment]
I was running 3.1.1.  Your fix worked although I had to install net-ssh 3.0.2.  The 3.0.1 version would make oxidized crash and complain it needs net-ssh >=3.0.2.  I tried 3.1.0 just to see if I could get that one to work but it had the same issue as 3.1.1.  

Thanks!
